src/actions/AppActions.ts

```diff
@@ -235,18 +236,32 @@ export interface SandboxRunAction extends AppAction {
 }
 
 export async function runTask(name: string, optional: boolean = false) {
+  // Runs the provided source in our fantasy gulp context
   const run = async (src: string) => {
     const gulp = new Gulpy();
     const project = appStore.getProject().getModel();
-    const context = {
+    contextify(src,
+      // thisArg
+      gulp,
+    {
+      // context for backwards compatibility
       gulp,
-      project,
       Service,
-      Language,
+      project,
       logLn,
-      filetypeForExtension,
-    };
-    Function.apply(null, Object.keys(context).concat(src)).apply(gulp, Object.values(context));
+      filetypeForExtension
+    }, {
+      // modules
+      "gulp": gulp,
+      "@wasm/studio-utils": {
+        Service,
+        project,
+        logLn,
+        filetypeForExtension,
+        setStatus,
+        clearStatus
```

Comment:
Some of these might not be in the corresponding package, but that's what I use as an example.

Reply:
Personally, I would like to avoid that custom templates can control UI. Can we remove setStatus/clearStatus and their usage from setup.js?